airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py

```diff
@@ -108,17 +126,109 @@ def record_message_from_record(record_: Dict) -> List[AirbyteMessage]:
             )
 
 
-def get_default_catalog(schema):
-    return ConfiguredAirbyteCatalog(
-        streams=[
-            ConfiguredAirbyteStream(
-                stream=AirbyteStream(
-                    name="test_stream",
-                    json_schema=schema,
-                    supported_sync_modes=["full_refresh"],
-                ),
-                sync_mode="full_refresh",
-                destination_sync_mode="overwrite",
+recordset_comparison_test_cases = [
+    (
+        [["id"]],
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        [{"id": 1, "first_name": "Albert", "last_name": "Einstein"}, {"id": 2, "first_name": "Joseph", "last_name": "Lagrange"}],
+        False,
+        "pk_sets_equal_success",
+    ),
+    (
```

Comment:
```suggestion
from contextlib import nullcontext as does_not_raise

recordset_comparison_test_cases = [
    (
        [["id"]],
        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
        [{"id": 1, "first_name": "Albert", "last_name": "Einstein"}, {"id": 2, "first_name": "Joseph", "last_name": "Lagrange"}],
        does_not_raise(),
        "pk_sets_equal_success",
    ), 
      (
        [["id"]],
        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
        [{"id": 2, "first_name": "Joseph", "last_name": "Lagrange"}],
        pytest.raises(Failed),
        "pk_sets_equal_success",
    ),
```

And then `if should_fail` is not required.

Reply:
nice solution, thank you!